backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5629440_sys_gh12640_matchinvoice_window_improvement.sql

```diff
@@ -0,0 +1,53 @@
+CREATE TABLE backup.m_matchinv_gh11167_20220311 as SELECT * FROM m_matchinv;
+
+update m_matchinv  set c_invoice_id = (select il.c_invoice_id 
+                                       from c_invoiceline il 
+                                       where il.c_invoiceline_id = m_matchinv.c_invoiceline_id)
+where c_invoiceline_id is not null;
+
+
+update m_matchinv  set m_inout_id= (select iol.m_inout_id 
+                                    from m_inoutline iol
+                                    where iol.m_inoutline_id = m_matchinv.m_inoutline_id)
+where m_inoutline_id  is not null;
+
+
+create index idx_m_matchinv_m_inout_id 
+on
+m_matchinv(m_inout_id);
+
+
+create index idx_m_matchinv_c_invoice_id 
+on
+m_matchinv(c_invoice_id);
+
+-- 2022-03-10T13:04:46.942Z
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+UPDATE AD_Column SET IsMandatory='Y',Updated=TO_TIMESTAMP('2022-03-10 14:04:46','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Column_ID=582453
```

Comment:
front this line (inclusive) to the end, you have to move it to a new next script.

e..g. This script is called `5629440_sys_gh12640_matchinvoice_window_improvement.sql`.
U can create a new one called `5629441_sys_sys_gh12640_Make_Invoice_and_InOut_mandatory.sql`.

## why?
because in PostgreSQL u cannot mix DDL and DML operations on a table without a "COMMIT" in between.
But I discourage having "COMMIT;" inside our scripts. 
So I strongly suggest moving it to a new script.